forte/processors/base/index_processor2.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Index processor2, with original datapack information
+"""
+from abc import ABC
+from typing import Dict, Any, List, Tuple
+
+from forte.common import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.processors.base.pack_processor import PackProcessor
+
+
+__all__ = [
+    "IndexProcessor2"
+]
+
+
+class IndexProcessor2(PackProcessor, ABC):
```

Comment:
First, just put this into `index_processor.py`. And second, `IndexProcessor2` is not a good name, please rename it to be more meaningful.